src/texture.rs

```diff
@@ -58,8 +58,10 @@ impl Debug for PerlinTexture {
     }
 }
 
+unsafe impl Sync for PerlinTexture{}
```

Comment:
Instead of manually `impl`ementing an unsafe trait you could just adjust your definition of `PerlinTexture`:
```
pub struct PerlinTexture {
    noise: Box<dyn Fn(V3, f64) -> f64 + Sync>,
    scale: f64,
}
```